oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Document.java

```diff
@@ -151,6 +151,22 @@ public void seal() {
         }
     }
 
+    /**
+     * UnSeals this document and turns it into a mutable object.
+     *
+     * @see #seal()
+     */
+    void unSeal() {
```

Comment:
I don't like adding method. The idea behind sealing a document is to ensure it becomes immutable. A DocumentStore implementation can use this to construct a Document, put it into a cache and then return it to a caller. With this new method a caller can modify a sealed Document, violating a basic principle.